pkg/action/archive.go

```diff
@@ -140,6 +142,27 @@ func extractTar(ctx context.Context, d string, f string) error {
 		return nil
 	case strings.Contains(filename, ".bz2") || strings.Contains(filename, ".bzip2"):
 		br := bzip2.NewReader(tf)
+		// Extract non-tar archives directly
```

Comment:
should code be located outside of the `extractTar` method then?

If you made this new code it's own method, it'd remove the complexity.

Reply:
Changed in [`c32840a` (#702)](https://github.com/chainguard-dev/malcontent/pull/702/commits/c32840af4cba55983f37e6e233c37170267ba06b).